crates/codegen/src/verilog/memory.rs

```diff
@@ -0,0 +1,99 @@
+//! Transforms the graph to use memory / flip flops
+//! Replaces the root func node with loads from memory
+//! Replaces the leaf call nodes with stores to memory
+
+use std::collections::BTreeMap;
+
+use crate::*;
+use tohdl_ir::expr::*;
+use tohdl_ir::graph::DataFlow;
+use tohdl_ir::graph::*;
+use tohdl_passes::Transform;
+use tohdl_passes::TransformResultType;
+
+#[derive(Clone, PartialEq)]
+pub struct NextStateNode {
+    pub values: Vec<Expr>,
+}
+
+impl std::fmt::Display for NextStateNode {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let buf = self
```

Comment:
Unsure about convention, but I prefer functions on new lines, but accessing on the same line, so that you know what the functions are "acting" on. I.e.
```rust
self.values
    .iter()
...
```
Shows that all latter functions are acting on `self.values` rather than `self`

Reply:
I'll setup a auto-formatter in #18.